File: week_1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_positive_int(void);

int main(void)
{
    int change = 0;

    int change_owed = get_positive_int();

    int quarters = change_owed / 25;
    change_owed -= (25 * quarters);

    int dimes = change_owed / 10;
    change_owed -= (10 * dimes);

    int nickels = change_owed / 5;
    change_owed -= (5 * nickels);

    change = quarters + dimes + nickels + change_owed;

    printf("%i\n", change);
}

int get_positive_int(void)
{
    int n;
    do
    {
        n = get_int("Change owed: ");
    }
    while (n < 1);
    return n;
}

File: week_6/cash.py
def get_change():
    while True:
        try:
            change = float(input("Change: "))
            if change > 0:
                return change
        except ValueError:
            pass

change_owed = get_change()
change_owed *= 100

quarters = change_owed // 25
change_owed -= (25 * quarters)

dimes = change_owed // 10
change_owed -= (10 * dimes)

nickels = change_owed // 5
change_owed -= (5 * nickels)

change = quarters + dimes + nickels + change_owed

print(int(change))
